### PR TITLE
[FEATURE] Add Laravel as application

### PR DESCRIPTION
**Pull request by simonschaufi**
- [x] Tests for the changes have been added (for bug fixes / features)

The workflow is exactly the same like in deployer: https://github.com/deployphp/deployer/blob/c3d236a81f35e26592c8fa51b5f81639cd52db7c/recipe/laravel.php

### CODE REVIEW

**Comment by t3easy** [comment]
Sorry, I have no experience with Laravel so I don't know what steps are nessesary to deploy an app. Someone else have to review.

**Comment by simonschaufi** [comment]
I use that code in several projects already and just extracted it to the core. Still it's good to have someone else have a look at it but it will be hard to find a laravel developer quickly as surf is not popular enough

**Comment by t3easy** [comment]
Maybe reference to a common deloyer workflow?

**Comment by simonschaufi** [comment]
There are lots of resources and this is one of them: https://github.com/lorisleiva/laravel-deployer

File: src/Task/Laravel/ClearAuthResetsTask.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+     * @param Node $node
+     * @param Application $application
+     * @param Deployment $deployment
+     * @param array $options
+     * @throws TaskExecutionException
+     * @throws InvalidConfigurationException
+     */
+    public function execute(Node $node, Application $application, Deployment $deployment, array $options = [])
+    {
+        if (!$application instanceof Laravel) {
```
Comment (sabbelasichon) [inline src/Task/Laravel/ClearAuthResetsTask.php]: We should use the Assertion from the webmozart/assert library here.

**Comment by simonschaufi** [comment]
@sabbelasichon what kind of tests should I add?

**Comment by sabbelasichon** [comment]
If you like, feel free to merge it whenever you want

**Comment by sabbelasichon** [comment]
@simonschaufi If you like merge it and we give it a try.

**Comment by simonschaufi** [comment]
I'm currently working on it, I need to add some tests also...

**Comment by simonschaufi** [comment]
@sabbelasichon :partying_face:

**Pull request merged**